app/assets/javascripts/topology/graph/editor_graph.js

```diff
@@ -0,0 +1,129 @@
+/*globals EditorGraph,Topology*/
+
+Topology.EditorGraph = (function () {
+    'use strict';
+
+    function plusPath(size) {
+        return 'M0,-' + size + ' V' + size +' M-' + size + ',0 H' + size;
+    }
+
+    EditorGraph.prototype = $.extend({}, Topology.Base.prototype, {
+        draw: function () {
+            this.buildBaseSVG();
+
+            this.svgGroup.attr('transform', "translate(0, 40)");
+
+            this.nodeIds = this.maxId();
+            this.tree    = d3.layout.tree().nodeSize([80, 0]);
+
+            this.update();
+
+            this.center(this.data);
+            window.TopologyEditor.form.show(this.data);
+        },
+
+        update: function () {
+            var node,
+                nodeEnter,
+                link,
+                addSize = 6,
+                nodes   = this.tree.nodes(this.data),
+                links   = this.tree.links(nodes);
+
+            nodes.forEach(function (d) {
+                d.y = d.depth * this.lineSpace;
+            }.bind(this));
+
+            this.svgGroup.selectAll("g.node, path.link").remove();
+
+            node = this.svgGroup.selectAll("g.node").data(nodes, function (d) {
+                return d.id || (d.id = ++this.nodeIds);
+            }.bind(this));
+
+            nodeEnter = node.enter().append("g")
+                .attr("class", function (d) {
+                    if (d.focus) {
+                        window.TopologyEditor.graphEditor.focusId = d.id;
+                        return "node focus";
+                    } else {
+                        return "node";
+                    }
+                })
+                .attr("transform", function (d) {
+                    return "translate(" + d.x + "," + d.y + ")";
+                });
+
+            // append a + button
+            nodeEnter.append("path")
+                .attr("class", "add-button")
+                .attr('d', plusPath(addSize))
+                .attr('stroke', 'green')
+                .attr('stroke-width', 3)
+                .attr('fill', 'none')
+                .attr('transform', 'translate(22, 0)')
+                .on('click', function (d) {
+                    window.TopologyEditor.graphEditor.addNode(d);
+                })
+                .on("mouseover", function () {
```

Comment:
Hmm I think I stumbled upon the same issue as you:

> makes the button frustratingly difficult to click

But didn't think it were my clicking skills and went for this farfetched solution. Yes the `cursor: pointer`  in the SASS just works. 😅